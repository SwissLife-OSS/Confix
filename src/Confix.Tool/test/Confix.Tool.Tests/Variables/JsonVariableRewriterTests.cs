using System.Text.Json.Nodes;
using ConfiX.Variables;
using Snapshooter.Xunit;

namespace Confix.Tool.Tests;

public class JsonVariableRewriterTests
{
    [Fact]
    public void Rewrite_ShouldReplaceAllVars()
    {
        // arrange
        JsonNode? node = JsonNode.Parse("""
            {
                "foo": {
                    "bar": "baz",
                    "test": "$test:variable.number"
                },
                "bar": ["$test:variable.string"],
                "number": 42,
                "bool": "$test:variable.bool"
            }
        """);
        var variableLookup = new Dictionary<VariablePath, JsonValue> {
            { VariablePath.Parse("$test:variable.string"), JsonValue.Create("someReplacedValue")},
            { VariablePath.Parse("$test:variable.number"), JsonValue.Create(420)},
            { VariablePath.Parse("$test:variable.bool"), JsonValue.Create(true)},
        };

        // act
        var result = new JsonVariableRewriter(variableLookup).Rewrite(node);

        // assert
        result?.ToString().MatchSnapshot();
    }
}
