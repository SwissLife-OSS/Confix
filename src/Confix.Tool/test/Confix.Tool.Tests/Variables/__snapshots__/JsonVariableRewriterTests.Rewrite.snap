{"foo":{"bar":"baz","test":"someReplacedValue"},"bar":["someReplacedValue"],"number":42}
